zcash_primitives/src/transaction/builder.rs

```diff
@@ -223,6 +225,7 @@ impl TransparentInputs {
                 consensus_branch_id,
                 SIGHASH_ALL,
                 Some((i, &info.coin.script_pubkey, info.coin.value)),
+                // tze equivalent is ???
```

Comment:
I think the approach we will end up taking is to serialize the TZE data in place of the transparent data, so this would become `Option<InputData>` with:
```rust
enum InputData {
    Transparent {
        index: usize,
        script_pubkey: Script,
        value: Amount,
    },
    Tze {
        index: usize,
        precondition: Vec<u8>,
        value: Amount,
    },
}
```
and we would domain-separate the transparent and TZE data types in the signature hash.

Reply:
This is the one piece that I think we should perhaps pair on; does this require a change to the ZIP?

Reply:
Yes, the ZIP will need to be updated to describe how it changes ZIP 243.